source/common/http/conn_pool_grid.cc

```diff
@@ -99,17 +110,37 @@ void ConnectivityGrid::WrapperCallbacks::onConnectionAttemptReady(
     ConnectionAttemptCallbacks* attempt, RequestEncoder& encoder,
     Upstream::HostDescriptionConstSharedPtr host, const StreamInfo::StreamInfo& info,
     absl::optional<Http::Protocol> protocol) {
+  ASSERT(host == grid_.host_);
```

Comment:
you probably already know this, but I'll point it out once because it really surprised me the first time, Envoy `ASSERT` is like `DCHECK`, it's a no-op in release builds.

Reply:
Yup! *super* surprising but ok :) (I'd be inclined to rename ASSERT => DEBUG_ASSERT and rename RELEASE_ASSERT => ASSERT, but c'est la vie)